ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -247,90 +246,54 @@ def empty(spec: "BehaviorSpec") -> "TerminalSteps":
 class ActionType(Enum):
     DISCRETE = 0
     CONTINUOUS = 1
+    HYBRID = 2
 
 
-class BehaviorSpec(NamedTuple):
-    """
-    A NamedTuple to containing information about the observations and actions
-    spaces for a group of Agents under the same behavior.
-     - observation_shapes is a List of Tuples of int : Each Tuple corresponds
-     to an observation's dimensions. The shape tuples have the same ordering as
-     the ordering of the DecisionSteps and TerminalSteps.
-     - action_type is the type of data of the action. it can be discrete or
-     continuous. If discrete, the action tensors are expected to be int32. If
-     continuous, the actions are expected to be float32.
-     - action_shape is:
-       - An int in continuous action space corresponding to the number of
-     floats that constitute the action.
-       - A Tuple of int in discrete action space where each int corresponds to
-       the number of discrete actions available to the agent.
-    """
-
-    observation_shapes: List[Tuple]
-    action_type: ActionType
-    action_shape: Union[int, Tuple[int, ...]]
+class ActionSpec(NamedTuple):
+    num_continuous_actions: int
+    discrete_branch_sizes: Tuple[int, ...]
 
+    # For backwards compatibility
     def is_action_discrete(self) -> bool:
         """
         Returns true if this Behavior uses discrete actions
         """
-        return self.action_type == ActionType.DISCRETE
+        return self.discrete_action_size > 0
```

Comment:
I think it would make more sense if this was 
```suggestion
        return self.discrete_action_size > 0 and self.continuous_action_size == 0
```

Reply:
I think it might be best to raise an exception in this and is_action_discrete and if `self.discrete_action_size > 0 and self.continuous_action_size > 0` - basically if you have a hybrid action space, this is asking the wrong question.

Reply:
This is mostly for backward compatibility and could be removed after the hybrid action spaces are merged. Additionally, I raise in `policy.py` in this PR if theres a hybrid action spec.

Reply:
I think everything that will be removed should be flagged in the docstrings. We introduce a lot of new APIs in this PR and we need to remove them at some point.